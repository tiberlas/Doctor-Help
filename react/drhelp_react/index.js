const express = require('express');
const app = express();

require('./routes')(app); //these are your api routes

//Non api requests in production
if (process.env.NODE_ENV === 'production' || process.env.NODE_ENV === 'staging') {
    // Add production middleware such as redirecting to https

    // Express will serve up production assets i.e. main.js
    app.use(express.static(__dirname + '/client/build'));
    // If Express doesn't recognize route serve index.html
    const path = require('path');
    app.get('*', (req, res) => {
        res.sendFile(
            path.resolve(__dirname, 'client', 'build', 'index.html')
        );
    });
}

//start server
const PORT = process.env.PORT || 5000; //Heroku sets port dynamically
app.listen(PORT, () => {
    console.log('listening...');
}).on('error', err => {
    console.log(`Error Code: ${err.code}`);
});