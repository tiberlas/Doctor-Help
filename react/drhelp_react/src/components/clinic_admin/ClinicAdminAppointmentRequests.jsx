import React, { Component } from "react";
import { ClinicAdminContext } from "../../context/ClinicAdminContextProvider";
import axios from "axios";
import Table from "@material-ui/core/Table";
import TableBody from "@material-ui/core/TableBody";
import TableCell from "@material-ui/core/TableCell";
import TableHead from "@material-ui/core/TableHead";
import TableRow from "@material-ui/core/TableRow";
import RequestedAppointmentItem from "../requesting_appointment/RequestedAppointmentItem.jsx";

class ClinicAdminAppointmentRequests extends Component {
	state = {
		appointments: [],
		name: "",
	};

	static contextType = ClinicAdminContext;

	componentDidMount() {
		this.handleGetRequests();
		this.handleClinicName();
	}

	handleGetRequests = () => {
		axios
			.get("http://localhost:8080/api/appointments/requests/all")
			.then((response) => {
				this.setState({ appointments: response.data });
			});
	};

	handleClinicName = () => {
		axios
			.get(
				"http://localhost:8080/api/clinics/id=" +
					this.context.admin.clinicId,
			)
			.then((response) => {
				this.setState({
					name: response.data.name,
				});
			});
	};

	handelUpdate = () => {};

	render() {
		let i = 0;
		return (
			<div class="row d-flex justify-content-center">
				<div class="col-md-7">
					<br />
					<h3>Clinic {this.state.name}</h3>
					<h4>List of requested appointments</h4>
					<br />
					<Table class="table table-hover ">
						<TableHead class="table-active">
							<TableRow class="table-active">
								<TableCell class="text-success cursor-pointer">
									date and time
								</TableCell>
								<TableCell class="text-success cursor-pointer">
									procedure
								</TableCell>
								<TableCell class="text-success cursor-pointer">
									doctor
								</TableCell>
								<TableCell class="text-success cursor-pointer">
									nurse
								</TableCell>
								<TableCell class="text-success cursor-pointer">
									patient
								</TableCell>
							</TableRow>
						</TableHead>
						<TableBody>
							{this.state.appointments.map((c) => (
								<TableRow
									className={++i % 2 ? `table-dark` : ``}
								>
									<RequestedAppointmentItem
										key={c.id}
										id={c.id}
										value={c}
										handleUpdate={this.handleUpdate}
									/>
								</TableRow>
							))}
						</TableBody>
					</Table>
				</div>
			</div>
		);
	}
}

export default ClinicAdminAppointmentRequests;
