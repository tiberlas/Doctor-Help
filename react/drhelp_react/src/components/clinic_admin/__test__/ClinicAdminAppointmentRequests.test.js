import React from "react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import { render, cleanup } from "@testing-library/react";
import ClinicAdminAppointmentRequests from "../ClinicAdminAppointmentRequests.jsx";
import ClinicAdminContextProvider from "../../../context/ClinicAdminContextProvider";
import { ExpansionPanelActions } from "@material-ui/core";

const admin = {
	id: 1,
	firstName: "BORKO",
	lastName: "MOTORKO",
	address: "kineska cetvrt 16",
	state: "bog te pita",
	city: "上海",
	phoneNumber: "064882553",
	email: "borkomotorko16@gamil",
	birthday: "03/06/1993",
	clinicId: 1,
};

afterEach(cleanup);
test("renders without crashing", () => {
	const div = document.createElement("div");

	ReactDOM.render(
		<ClinicAdminContextProvider admin={admin}>
			<ClinicAdminAppointmentRequests />
		</ClinicAdminContextProvider>,
		div,
	);
	ReactDOM.unmountComponentAtNode(div);
});

test("matches snapshot", () => {
	const ClinicAdminAppointemntRequest = renderer
		.create(
			<ClinicAdminContextProvider admin={admin}>
				<ClinicAdminAppointmentRequests />
			</ClinicAdminContextProvider>,
		)
		.toJSON();

	expect(ClinicAdminAppointemntRequest).toMatchSnapshot();
});
