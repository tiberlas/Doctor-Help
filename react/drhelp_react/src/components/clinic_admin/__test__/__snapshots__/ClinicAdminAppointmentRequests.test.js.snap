// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  class="row d-flex justify-content-center"
>
  <div
    class="col-md-7"
  >
    <br />
    <h3>
      Clinic 
      
    </h3>
    <h4>
      List of requested appointments
    </h4>
    <br />
    <table
      class="table table-hover "
      className="MuiTable-root"
    >
      <thead
        class="table-active"
        className="MuiTableHead-root"
      >
        <tr
          class="table-active"
          className="MuiTableRow-root MuiTableRow-head"
        >
          <th
            aria-sort={null}
            class="text-success cursor-pointer"
            className="MuiTableCell-root MuiTableCell-head"
            scope="col"
          >
            date and time
          </th>
          <th
            aria-sort={null}
            class="text-success cursor-pointer"
            className="MuiTableCell-root MuiTableCell-head"
            scope="col"
          >
            procedure
          </th>
          <th
            aria-sort={null}
            class="text-success cursor-pointer"
            className="MuiTableCell-root MuiTableCell-head"
            scope="col"
          >
            doctor
          </th>
          <th
            aria-sort={null}
            class="text-success cursor-pointer"
            className="MuiTableCell-root MuiTableCell-head"
            scope="col"
          >
            nurse
          </th>
          <th
            aria-sort={null}
            class="text-success cursor-pointer"
            className="MuiTableCell-root MuiTableCell-head"
            scope="col"
          >
            patient
          </th>
        </tr>
      </thead>
      <tbody
        className="MuiTableBody-root"
      />
    </table>
  </div>
</div>
`;
